src/Service.GraphQLBuilder/Sql/SchemaConverter.cs

```diff
@@ -31,184 +31,387 @@ public static class SchemaConverter
+        {
+            if (GraphQLUtils.CreateAuthorizationDirectiveIfNecessary(
+                                            roles,
+                                            out DirectiveNode? authZDirective))
+            {
+                directives.Add(authZDirective!);
+            }
+
+            string exposedColumnName = columnName;
+            if (configEntity.Mappings is not null && configEntity.Mappings.TryGetValue(key: columnName, out string? columnAlias))
+            {
+                exposedColumnName = columnAlias;
+            }
+
+            NamedTypeNode fieldType = new(GetGraphQLTypeFromSystemType(column.SystemType));
+            FieldDefinitionNode field = new(
+                location: null,
+                new(exposedColumnName),
+                description: null,
+                new List<InputValueDefinitionNode>(),
+                column.IsNullable ? fieldType : new NonNullTypeNode(fieldType),
+                directives);
+            return field;
+        }
+
+        /// <summary>
+        /// Helper method to generate field for a relationship for an entity. While processing the relationship, it does some other things:
+        /// 1. Helps in keeping track of relationships with cardinality M:N as whenever such a relationship is encountered,
+        /// the (soure, target) pair of entities is added to the collection of entities with many to many relationship.
+        /// 2. Helps in keeping track of fields from the source entity which hold foreign key references to the target entity.
+        /// </summary>
+        /// <param name="entityName">Name of the entity in the runtime config to generate the GraphQL object type for.</param>
+        /// <param name="databaseObject">SQL database object information.</param>
+        /// <param name="entities">Key/Value Collection mapping entity name to the entity object, currently used to lookup relationship metadata.</param>
+        /// <param name="isNestedMutationSupported">Whether nested mutation is supported for the entity.</param>
+        /// <param name="entitiesWithManyToManyRelationships">Collection of (source, target) entities which have an M:N relationship between them.</param>
+        /// <param name="foreignKeyFieldsInEntity">Set of fields from source entity holding foreign key references to a target entity.</param>
+        /// <param name="relationshipName">Name of the relationship.</param>
+        /// <param name="relationship">Relationship data.</param>
+        private static FieldDefinitionNode GenerateFieldForRelationship(
+            string entityName,
+            DatabaseObject databaseObject,
+            RuntimeEntities entities,
+            bool isNestedMutationSupported,
+            HashSet<Tuple<string, string>>? entitiesWithManyToManyRelationships,
```

Comment:
updating entitiesWithManyToManyRelationships as a side-effect of this function is not intuitive, unless the summary is read or implementation of function is understood, can we rename the function to better reflect what all its doing?

Reply:
Per discussion over call, added a method `Entity. GetSourceAndTargetEntityNameFromLinkingEntityName()` to get the source and target entity names from the linking entity name. 

Now, there is no need to populate the `entitiesWithManyToManyRelationships` collection. Hence removed it and the logic to populate it.